contribute/design/cstor-operator/specs.md

```diff
@@ -0,0 +1,235 @@
+
+1. Handle basic day 2 operations.
+2. More granular control and visibilty on specs to user while provisioning pool.
+3. Enhanced reconciliation loop for the CStor-operator.
+
+### Proposals
+
+#### Rename StoragePoolClaim to CStorPoolCluster
+
+Generally, claim resources maps/bounds to a single object, for example PVC i.e.
+PersistentVolumeClaim bounds to a PV object. We can say, claim resources has
+a one to one mapping with some object.
+
+But StoragePoolClaim is mapped to a set of cStor pools which can create confusion
+and does not look intutive from users prespective.
+
+CStorPoolCluster looks intutive i.e. CStorPoolCluster is logically a resource
+which is parent of a collection of CStorPools.
+
+
+#### Spec for CStorPoolCluster
+
+Following are few sample manual CStorPoolCluster specs
+
+1.
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-striped
+spec:
+  nodes:
+  - name: gke-cstor-it-default-pool-569eb31d-3l88
+    poolSpec:
+      poolType: striped
+    groups:
+    - name: group1
+      disks:
+        // Can use block device name here. In YAML spec 
+        // disks and block device will mean the same.
+      - name: sparse-3c1fc7491f9e4cf50053730740647318
+        id: disk1
+  name: cstor-pool-striped
+
+```
+
+2.
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-striped
+spec:
+  nodes:
+  - name: gke-cstor-it-default-pool-1
+    poolSpec:
+      poolType: striped
+    groups:
+    - name: group1
+      disks:
+      - name: sparse-pool-1-disk-1
+        id: pool-1-disk1
+      - name: sparse-pool-1-disk-2
+        id: pool-1-disk2
+  - name: gke-cstor-it-default-pool-2
+    poolSpec:
+      poolType: striped
+    groups:
+    - name: group1
+      disks:
+      - name: sparse-pool-2-disk-1
+        id: pool-2-disk1
+  name: cstor-pool-striped
+
+```
+
+3.
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-mirrored
+spec:
+  nodes:
+
+  - name: gke-cstor-it-default-pool-1
+    poolSpec:
+      poolType: mirrored
+
+    groups:
+    - name: group1
+      disks:
+      - name: sparse-pool-1-disk-1
+        id: pool-1-disk1
+
+      - name: sparse-pool-1-disk-2
+        id: pool-1-disk2
+
+    - name: group2
+      disks:
+      - name: sparse-pool-1-disk-3
+        id: pool-1-disk3
+
+      - name: sparse-pool-1-disk-4
+        id: pool-1-disk4
+
+  - name: gke-cstor-it-default-pool-2
+    poolSpec:
+      poolType: striped
+
+    groups:
+    - name: group1
+      disks:
+      - name: sparse-pool-2-disk-1
+        id: pool-2-disk1
+
+      - name: sparse-pool-2-disk-2
+        id: pool-2-disk2
+
+  name: cstor-pool-mirroed
+
+```
+4.
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-hybrid
+spec:
+  nodes:
+
+  - name: gke-cstor-it-default-pool-1
+    poolSpec:
+      poolType: mirrored
```

Comment:
can we explain in yaml comments, where is poolType applicable. e.g. against `nodes[i].groups` or against `nodes[i].groups[j].disks`